### PR TITLE
[ci] Add GPL options

### PR DESCRIPTION
**Pull request by dpiparo**
since no release is in sight to be able to have the same PR and nightly builds.

### CODE REVIEW

**Review by guitargeek — APPROVED** [review]
Great, thanks a lot! We should just not lose the overview in which branch we do what :slightly_smiling_face:

**Comment by pcanal** [comment]
I would have many those (GPL/non-GPL) part of the matrix.  We do *have to* make sure that a controlled sub-set of ROOT is working without any GPL libraries.

**Comment by dpiparo** [comment]
@pcanal has a point. We should also understand what to do with builtins: is it better to cater users that want drop in binaries with all externals, or completely rely on distro packages? Resources are not infinite and perhaps a choice has to be made.

**Comment by dpiparo** [comment]
> Great, thanks a lot! We should just not lose the overview in which branch we do what 🙂

6.26 and 6.28 always were including GSL before the move to GH CI. 6.30 was released w/o already. 6.32/master are not released and give us some space to maneuver.

**Review by lmoneta — APPROVED** [review]
LGTM!

**Review by lmoneta — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -2,55 +2,48 @@
...
-evaluation, minimization, graphics and visualization classes to allow
-the easy setup of an analysis system that can query and process the data
+ROOT is a unified software package for the storage, processing, visualization 
+and analysis of scientific data. It is reliable, performant and well supported,
+easy to use and obtain, and strives to minimize the computing resources needed 
+to achieve scientific results.
+
+ROOT is a comprehensive toolbox to process and analyze data, from its acquisition, 
+to its processing, analysis and final visualization in form of highly customizable, 
+publication-ready plots.
```
Comment (vgvassilev) [inline README.md:12]: Does that not repeat the previous paragraph?

File: README.md [hunk 2]
```diff
@@ -2,55 +2,48 @@
...
+full advantage of multi-core and distributed systems.
 
-Thanks to the built-in C++ interpreter cling, the command, the
-scripting and the programming language are all C++. The interpreter
-allows for fast prototyping of the macros since it removes the time
-consuming compile/link cycle. It also provides a good environment to
-learn C++. If more performance is needed the interactively developed
-macros can be compiled using a C++ compiler via a machine independent
-transparent compiler interface called ACliC.
+ROOT is written in C++, and, thanks to PyROOT, provides a powerful Python interface
```
Comment (vgvassilev) [inline README.md:23]: Can we ack `cppyy` as it relates to PyROOT in the same way as Cling to ROOT?

Comment (dpiparo) [reply README.md:23]: make a proposal?

File: README.md (hunk 1 above)
Comment (dpiparo) [reply README.md:12]: In a sense yes.

File: README.md [hunk 3]
```diff
@@ -2,55 +2,48 @@
...
 When citing ROOT, please use both the reference reported below and the DOI specific to your ROOT version available [on Zenodo](https://zenodo.org/badge/latestdoi/10994345) [![DOI](https://zenodo.org/badge/10994345.svg)](https://zenodo.org/badge/latestdoi/10994345). For example, you can copy-paste and fill in the following citation:
 
     Rene Brun and Fons Rademakers, ROOT - An Object Oriented Data Analysis Framework,
     Proceedings AIHENP'96 Workshop, Lausanne, Sep. 1996,
     Nucl. Inst. & Meth. in Phys. Res. A 389 (1997) 81-86.
-    See also "ROOT" [software], Release vX.YY/ZZ, dd/mm/yyyy,
-    (Select the right link for your release here: https://zenodo.org/search?page=1&size=20&q=conceptrecid:848818&all_versions&sort=-version).
-
-## Live Demo for CERN Users
-[![](https://swanserver.web.cern.ch/swanserver/images/badge_swan_white_150.png)](http://cern.ch/swanserver/cgi-bin/go?projurl=https://github.com/cernphsft/rootbinder.git)
```
Comment (pcanal) [inline README.md:44]: The Swan demo seems to still work (the binder do seem to no longer work i.e. need some update).  What is the benefit of removing the link?

Comment (dpiparo) [reply README.md:44]: That seemed to be too CERN-centric. I might have been overzealous though: do you think it can be re-added?

File: README.md [hunk 4]
```diff
@@ -2,55 +2,48 @@
...
-the easy setup of an analysis system that can query and process the data
+ROOT is a unified software package for the storage, processing, visualization 
+and analysis of scientific data. It is reliable, performant and well supported,
+easy to use and obtain, and strives to minimize the computing resources needed 
+to achieve scientific results.
+
+ROOT is a comprehensive toolbox to process and analyze data, from its acquisition, 
+to its processing, analysis and final visualization in form of highly customizable, 
+publication-ready plots.
+First of all, ROOT provides a very efficient storage system for data models,
```
Comment (jblomer) [inline README.md:13]:
```suggestion
ROOT provides a very efficient storage system for data models, 
```

File: README.md [hunk 5]
```diff
@@ -2,55 +2,48 @@
...
+easy to use and obtain, and strives to minimize the computing resources needed 
+to achieve scientific results.
+
+ROOT is a comprehensive toolbox to process and analyze data, from its acquisition, 
+to its processing, analysis and final visualization in form of highly customizable, 
+publication-ready plots.
+First of all, ROOT provides a very efficient storage system for data models, 
+that demonstrated to scale at the Large Hadron Collider experiments: Exabytes 
+of scientific data are written in columnar ROOT format.
+Moreover, ROOT comes with histogramming capabilities in an arbitrary number of
```
Comment (jblomer) [inline README.md:16]:
```suggestion
ROOT comes with histogramming capabilities in an arbitrary number of 
```

File: README.md [hunk 6]
```diff
@@ -2,55 +2,48 @@
...
+This is possible thanks to the powerful built-in C++ interpreter, Cling, that allows, 
+among the other things, to create dynamic python bindings at runtime, a C++ interactive 
+prompt as well as rapid C++ prototyping.
 
-The system has been designed in such a way that it can query its databases
-in parallel on clusters of workstations or many-core machines. ROOT is
-an open system that can be dynamically extended by linking external
-libraries. This makes ROOT a premier platform on which to build data
-acquisition, simulation and data analysis systems.
+ROOT fully embraces the open-source philosophy, it's made with passion by its community
```
Comment (jblomer) [inline README.md:29]:
```suggestion
ROOT fully embraces open-source, it's made with passion by its community 
```

File: README.md [hunk 7]
```diff
@@ -2,55 +2,48 @@
...
-[![](https://swanserver.web.cern.ch/swanserver/images/badge_swan_white_150.png)](http://cern.ch/swanserver/cgi-bin/go?projurl=https://github.com/cernphsft/rootbinder.git)
-
-## Screenshots
-These screenshots shows some of the plots (produced using ROOT) presented when the Higgs boson discovery was [announced at CERN](http://home.cern/topics/higgs-boson):
-
-![CMS Data MC Ratio Plot](https://root.cern/gallery/higgs_plots/CMS_Data_MC_Ratio_Plot.png)
-
-![Atlas P0 Trends](https://root.cern/gallery/higgs_plots/Atlas_P0_Trends.png)
-
-See more screenshots on our [gallery](https://root.cern/gallery).
```
Comment (jblomer) [inline README.md:53]: I would keep at least the link to the gallery.

File: README.md (hunk 2 above)
Comment (vgvassilev) [reply README.md:23]: Something like:

ROOT is performance critical software written in C++ and enables rapid prototyping powered by a unique C++ compliant interpreter called Cling. Cling enables efficient C++ type introspection which is a building block of flexible rapid prototyping in C++ and efficient automatic bindings with Python. The PyROOT component offers a powerful Python interface leveraging the [cppyy technology](https://cppinterop.readthedocs.io/). Cppyy uses Cling to connect to Python's runtime providing efficient, on-demand C++/Python interoperability.

**Pull request merged**